ntoskrnl/io/iomgr/iofunc.c

```diff
@@ -2158,7 +2158,8 @@ NtQueryInformationFile(IN HANDLE FileHandle,
     if (PreviousMode != KernelMode)
     {
         /* Validate the information class */
-        if ((FileInformationClass >= FileMaximumInformation) ||
+        if ((FileInformationClass < 0) ||
```

Comment:
Isn't ISO C supposed to map, by default, enumerations to unsigned int? If so, that would make that check useless.

Reply:
Quite the opposite. `enum` types are `int`

Reply:
Actually...
6.7.2.2#4
> Each enumerated type shall be compatible with `char`, a signed integer type, or an unsigned integer type. The choice of type is implementation-defined, but shall be capable of representing the values of all the members of the enumeration. The enumerated type is incomplete until after the } that terminates the list of enumerator declarations

Reply:
Oh how I hate the C standard.